qiskit/pulse/library/parametric_pulses.py

```diff
@@ -37,68 +37,224 @@ class ParametricPulseShapes(Enum):
         new_supported_pulse_name = library.YourPulseWaveformClass
 """
 from abc import abstractmethod
-from typing import Any, Dict, Optional, Union
+from typing import Any, Dict, List, Tuple, Optional, Union
 
-import math
 import numpy as np
 
 from qiskit.circuit.parameterexpression import ParameterExpression
 from qiskit.pulse.exceptions import PulseError
-from qiskit.pulse.library import continuous
-from qiskit.pulse.library.discrete import gaussian, gaussian_square, drag, constant
 from qiskit.pulse.library.pulse import Pulse
 from qiskit.pulse.library.waveform import Waveform
+from qiskit.pulse.utils import lambdify_symbolic_pulse
+
+# Pulse module doesn't employ symengine. It doesn't automatically simply piecewise function
+# which causes performance issue in the lambda function.
+# In addition, there are several unsupported expression of boolean operation.
+# Thanks to Lambdify at subclass instantiation, the performance regression is not significant here.
+import sympy as sym
```

Comment:
This needs to be a runtime import to avoid an import performance regression when the `parametric_pulses` module gets imported.

Reply:
One thing to note -- `ParametricPulse.__init_subclass__` uses `lambdify_symbolic_pulse` which uses some `sympy` functions. So even if the sympy usage was shifted to look like it was a run time import, sympy would still get imported when this module is loaded because of the subclasses like `Gaussian` which are defined here (and imported in `qiskit.pulse`'s `__init__.py`). At the cost of some extra complexity, this sympy usage could be shifted from class definition to first usage.